sacred/commandline_options.py

```diff
@@ -8,12 +8,72 @@
 """
 
 import warnings
+from typing import Callable
+import inspect
+import re
 
+from sacred.run import Run
 from sacred.commands import print_config
 from sacred.settings import SETTINGS
 from sacred.utils import convert_camel_case_to_snake_case, get_inheritors
 
 
+CLIFunction = Callable[[str, Run], None]
+
+
+class CLIOption:
+    def __init__(
+        self,
+        apply_function: CLIFunction,
+        short_flag: str,
+        long_flag: str,
+        is_flag: bool,
+    ):
+
+        assert re.match(r"-\w", short_flag)
+        assert re.match(r"--[\w-]+", long_flag)
```

Comment:
```suggestion
        assert re.match(r"--[\w-]+\w$", long_flag)
```
I think long flags should not end in hyphens. This way it should be ensured.